AK/QuickSelect.h

```diff
@@ -0,0 +1,169 @@
+/*
+ * Copyright (c) 2023, the SerenityOS developers.
+ *
+ * SPDX-License-Identifier: BSD-2-Clause
+ */
+
+#pragma once
+
+#include <AK/Math.h>
+#include <AK/Random.h>
+#include <AK/StdLibExtras.h>
+
+namespace AK {
+// FIXME: Stole and adapted these two functions from `Userland/Demos/Tubes/Tubes.cpp` we really need something like this in `AK/Random.h`
+static inline double random_double()
+{
+    return get_random<u32>() / static_cast<double>(NumericLimits<u32>::max());
+}
+
+static inline int random_int(size_t min, size_t max)
+{
+    return min + round_to<size_t>(random_double() * (max - min));
+}
```

Comment:
Hi, author of the Tubes' variant here. We're returning an `int` but working with `size_t min` and `max`, that doesn't look right?

Reply:
Yeah, I think I had to mess with it to make type checking happy. I'll make it consistent in the evening today.